### PR TITLE
Aqa/js functions arrays

### PR DESCRIPTION
**Pull request by manatskaya**
Hi Luibov,
Please review when you have a chance. 
Thank you
Regards,
Svitlana

### CODE REVIEW

File: js-functions-arrays/functions-arrays.js [hunk 1]
```diff
@@ -0,0 +1,144 @@
+// Task 1.1. Function declaration:
+function myFunctionDeclaration (a,b) {
+    if (a > b) {
+        return a;
+    }
+    return b;
```
Comment (landriiuk) [inline js-functions-arrays/functions-arrays.js]: you can use the ternary operator to enhance your code.

 return a > b ? a : b;

File: js-functions-arrays/functions-arrays.js [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+let resultFunctionDeclaration = myFunctionDeclaration (5,10);
+
+console.log('result of function declaration',resultFunctionDeclaration);
+
+// Task 1.2. Function expression:
+let myFunctionExpression = function(a,b) {
+    if (a > b) {
+       return a;
+    }
+    return b;
```
Comment (landriiuk) [inline js-functions-arrays/functions-arrays.js]: same here

File: js-functions-arrays/functions-arrays.js [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+console.log('result of function expression', resultFunctionExpression);
+
+// 1.3. Arrow function, varian 1:
+let myFunctionArrow = (a, b) => {
+    if (a > b){ 
+        return a;
+    }
+    return b;
+}
+let myFunctionArrowResult = myFunctionArrow (8,3);
```
Comment (landriiuk) [inline js-functions-arrays/functions-arrays.js:26]: using the ternary operator, you will have only one code line.

File: js-functions-arrays/functions-arrays.js [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+        throw new Error('length of Array is too short');
+    }
+
+    try {
+        for (let i = 0; i < numbers.length; i++) {
+            let currentValue = numbers[i];
+            let res;
+            console.log(currentValue);
+            if (typeof currentValue !== 'number' || currentValue === 0) {
+              throw new Error;
```
Comment (landriiuk) [inline js-functions-arrays/functions-arrays.js]: add err message text

File: js-functions-arrays/functions-arrays.js [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+                res = currentValue / numbers[numbers.length - 1];
+            } else {
+                res = currentValue / numbers[i - 1]
+            }
+
+            result.push(res);
+        }
+    }
+    catch (error) { 
+        console.log("error message: this is new error your element in the Array contains not equal to number or zero");
```
Comment (landriiuk) [inline js-functions-arrays/functions-arrays.js]: The error object provides two useful properties: name and message.
pls use: error.message to show error

File: js-functions-arrays/functions-arrays.js [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+    if(numbers.length < 2) {
+        throw new Error('length of Array is too short');
+    }
+
+    try {
+        for (let i = 0; i < numbers.length; i++) {
+            let currentValue = numbers[i];
+            let res;
+            console.log(currentValue);
+            if (typeof currentValue !== 'number' || currentValue === 0) {
```
Comment (landriiuk) [inline js-functions-arrays/functions-arrays.js]:
```suggestion
            if (typeof currentValue !== 'number' || currentValue === 0 || numbers[i-1] === 0) {
```

File: js-functions-arrays/functions-arrays.js [hunk 7]
```diff
@@ -0,0 +1,144 @@
...
+// Task 4:
+// assumption made that the first element shall be divided on the last element in the array
+
+const numbers = [1, 2, 10, 3]; 
+function divideArray(numbers) {
+    const result = [];
+    
+    if(numbers.length < 2) {
+        throw new Error('length of Array is too short');
+    }
```
Comment (landriiuk) [inline js-functions-arrays/functions-arrays.js]: should be inside try block

**Comment by landriiuk** [comment]
@manatskaya there are few moments to improve

**Comment by manatskaya** [comment]
Hi Luibov,
Thank you for the feedback provided.
Updated according to your comments.
Svitlana

File: js-functions-arrays/functions-arrays.js [hunk 8]
```diff
@@ -34,8 +28,8 @@ let myFunctionArrowResult = myFunctionArrow (8,3);
 console.log('result of arrow function', myFunctionArrowResult);
 
 // 1.3. Arrow function, varian 2:
-let myFunctionArrowBoolean = (a, b) => a > b ? a : b;
-let myFunctionArrowResultSimple = myFunctionArrowBoolean (12,4);
+let myFunctionArrowBoolean = (a, b) => {return a > b ? a : b};
```
Comment (landriiuk) [inline js-functions-arrays/functions-arrays.js]: you don't need to use "return" if there is only 1 line with code. If in arrow func 1 line with code, you can omit {} and return

**Review by landriiuk — APPROVED** [review]

**Comment by landriiuk** [comment]
Added one enhancement. 
Pls add it, and then merge pr

**Comment by manatskaya** [comment]
Hi Luibov,
Thank you for your feedback. Done.
Svitlana

**Pull request merged**